UnityProject/Assets/Scripts/Electricity/ConnectionMap.cs

```diff
@@ -5,191 +5,98 @@ public static class ConnectionMap
 	{
+		Connection validDirection;
 		switch (ConnectionDirection)
 		{
 			case Connection.North:
-				{
-					if (AdjacentConnections.pointA == Connection.South || AdjacentConnections.pointB == Connection.South)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.South;
 				break;
+			}
 			case Connection.NorthEast:
-				{
-					if (AdjacentConnections.pointA == Connection.SouthWest || AdjacentConnections.pointB == Connection.SouthWest)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.SouthWest;
 				break;
+			}
 			case Connection.East:
-				{
-					if (AdjacentConnections.pointA == Connection.West || AdjacentConnections.pointB == Connection.West)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.West;
 				break;
+			}
 			case Connection.SouthEast:
-				{
-					if (AdjacentConnections.pointA == Connection.NorthWest || AdjacentConnections.pointB == Connection.NorthWest)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.NorthWest;
 				break;
+			}
 			case Connection.South:
-				{
-					if (AdjacentConnections.pointA == Connection.North || AdjacentConnections.pointB == Connection.North)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.North;
 				break;
+			}
 			case Connection.SouthWest:
-				{
-					if (AdjacentConnections.pointA == Connection.NorthEast || AdjacentConnections.pointB == Connection.NorthEast)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.NorthEast;
 				break;
+			}
 			case Connection.West:
-				{
-					if (AdjacentConnections.pointA == Connection.East || AdjacentConnections.pointB == Connection.East)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.East;
 				break;
+			}
 			case Connection.NorthWest:
-				{
-					if (AdjacentConnections.pointA == Connection.SouthEast || AdjacentConnections.pointB == Connection.SouthEast)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.SouthEast;
 				break;
+			}
 			case Connection.Overlap:
-				{
-					if (AdjacentConnections.pointA == Connection.Overlap || AdjacentConnections.pointB == Connection.Overlap)
-					{
-						return (true);
-					}
-				}
+			{
+				validDirection = Connection.Overlap;
 				break;
+			}
 			case Connection.MachineConnect:
-				{
-					if (AdjacentConnections.pointA == Connection.MachineConnect || AdjacentConnections.pointB == Connection.MachineConnect)
-					{
-						return (true);
-					}
-				}
-				break;
+			{
+				validDirection = Connection.MachineConnect;
+				break;
+			}
+			default:
+			{
+				return false;
+			}
 		}
-		return (false);
-	}
 
+		return (AdjacentConnections.pointA == validDirection || AdjacentConnections.pointB == validDirection);
+	}
 
 	public static bool IsConnectedToTileOverlap(Connection ConnectionDirection, ConnPoint AdjacentConnections)
```

Comment:
This is meant for overlapping connections  on a tile
like 

![image](https://user-images.githubusercontent.com/27101670/81122673-4c044200-8f29-11ea-8816-1954d1e13e92.png)


Should technically be still connected

Even though to a bug I introduced accidentallyThis doesn't work anymore,  since it expects connections to be a new you Hashset every time, so You have to do record if it got any connections the usual way, if not then check if there is overlap

Reply:
It seems to be empty so therefore these wouldn't connect

Reply:
First comment should be addressed in 1e62f26
Second in 85ab8fd